### DISCUSSION TITLE
How to get Label widget to automatically wrap for long text?

### DISCUSSION THREAD

**Discussion by learnbyexample** [Textualize/textual#1684] [Q&A] [answered]
Here's a sample app. `Static` is working as expected but I'm not able to make `Label` to wrap long text. Also, passing output of `Panel` to `Label` wraps around but the right box border is missing. Textual version is 0.10.0

```python
from textual.app import App
from textual.widgets import Label, Static
from rich.panel import Panel

class LabelWrap(App):
    CSS = '''Screen {
              align: center middle;
          }

          #l_data {
              border: blank;
              background: lightgray;
          }

          #s_data {
              border: blank;
              background: lightgreen;
          }

          #p_data {
              border: blank;
              background: lightgray;
          }'''

    def __init__(self):
        super().__init__()

        self.data = ('Apple Banana Cherry Mango Fig Guava Pineapple:'
                     'Dragon Unicorn Centaur Phoenix Chimera Castle')

    def compose(self):
        yield Label(self.data, id='l_data')
        yield Static(self.data, id='s_data')
        yield Label(Panel(self.data), id='p_data')

    def on_mount(self):
        self.dark = False

if __name__ == '__main__':
    app = LabelWrap()
    app.run()
```

Screenshot of running the above code with 72x18 terminal dimensions:
![label_wrap](https://user-images.githubusercontent.com/17766317/215263901-d2a4df3c-6fb2-4bda-b209-f9ea1c98eb9f.png)

**Comment by willmcgugan** [comment, marked as answer]
You can add `shrink=True` on to the Label with the panel to allow it to shrink beyond the optimal size, and let it wrap. 

But unless you need a Panel in particular (for titles), its generally easier to set the border rule...

**Reply by learnbyexample** [reply]
Adding `shrink=True` works with/without Panel, thanks! I did check https://textual.textualize.io/widgets/label/ but this option isn't mentioned.

Any particular reason wrapping text isn't the default?

**Reply by willmcgugan** [reply]
It's inherited from Static.  It's documented in the API reference, but its not very discoverable. We will need to document that a bit better.

It's less to do with wrapping and more to do with calculating the optimal width of content. I suspect there is a bug there though. I would have expected your original code to still fit the panel in the screen.

**Reply by learnbyexample** [reply]
Thanks, I did know Label is inherited from Static but hadn't thought to check under API. Will keep that in mind for future.

Good to know my attempts to get wrapped text in Label resulted in identifying a possible bug ;)
